src/Utilities/ToolLocationHelper.cs

```diff
@@ -3399,7 +3399,8 @@ internal static string ConvertDotNetFrameworkArchitectureToProcessorArchitecture
             switch (architecture)
             {
                 case DotNetFrameworkArchitecture.Bitness32:
-                    if (ProcessorArchitecture.CurrentProcessArchitecture == ProcessorArchitecture.ARM)
+                    if (ProcessorArchitecture.CurrentProcessArchitecture == ProcessorArchitecture.ARM ||
+                        ProcessorArchitecture.CurrentProcessArchitecture == ProcessorArchitecture.ARM64)
```

Comment:
It looks like the original purpose of `ConvertDotNetFrameworkArchitectureToProcessorArchitecture` was to make it possible to compose 32-bit paths on AMD64. The AMD64 .NET Framework environment is special in that it guarantees the presence of the x86 version of .NET Framework + SDKs.

I don't believe this applies to ARM, be it Core or Framework. In fact, `ProcessorArchitectures.ARM` is not a supported architecture at all and should be removed. Together with `IA64` 😉 

No harm in the way it is written now as the scenarios just don't exist but high-level I don't think the code makes sense. If you look at where the result of this method is used (outside of tests), you'll see it's very AMD64 / x86 centric. I wonder if other architectures (so only `ARM64` as everything else is dead) shouldn't be handled differently in the top-level ToolLocationHelper methods.

Just a nit/rant, totally OK to leave it as is.